paparazzi-preview-processor/src/main/java/app/cash/paparazzi/preview/processor/PreviewProcessor.kt

```diff
@@ -15,5 +19,52 @@ public class PreviewProcessorProvider : SymbolProcessorProvider {
 public class PreviewProcessor(
   private val environment: SymbolProcessorEnvironment
 ) : SymbolProcessor {
-  override fun process(resolver: Resolver): List<KSAnnotated> = emptyList()
+
+  override fun process(resolver: Resolver): List<KSAnnotated> {
+    // Due to codgen and multi-round processing of ksp
+    // https://kotlinlang.org/docs/ksp-multi-round.html
+    if (resolver.getAllFiles().any {
+        it.fileName.contains("PaparazziPreviews") && it.filePath.contains("ksp")
+      }
+    ) {
+      "Skipping subsequent run due to PaparazziPreviews.kt already created and caused ksp re-run".log()
+      return emptyList()
+    }
+
+    val allFiles = resolver.getAllFiles().toList()
+    if (allFiles.isEmpty()) return emptyList()
+
+    val env = EnvironmentOptions(
+      namespace = environment.options["app.cash.paparazzi.preview.namespace"]!!
+    )
+
+    val dependencies = Dependencies(true, *allFiles.toTypedArray())
+    val isTestSourceSet = env.discoverVariant(dependencies).endsWith("UnitTest")
+
+    return resolver.getSymbolsWithAnnotation("androidx.compose.runtime.Composable")
+      .findPaparazzi()
```

Comment:
Why not look for `@Paparazzi` first since 1) there will be at most equal, but more likely, many fewer instances of it vs `@Composable`?